cylc/flow/suite_files.py

```diff
@@ -976,8 +1003,44 @@ def get_rsync_rund_cmd(src, dst, restart=False):
     return rsync_cmd
 
 
+def reinstall_workflow(named_run, rundir, source, dry_run=False):
+    """ Reinstall workflow.
+
+    Args:
+        named_run (str):
+            name of the run e.g. my-flow/run1
+        rundir (path):
+            run directory
+        source (path):
+            source directory
+        dry_run (bool):
+            if True, will not exectute the file transfer but report what would
+            be changed.
+    """
+    validate_source_dir(source, named_run)
+    check_nested_run_dirs(rundir, named_run)
+    REINSTALL_LOG = _get_logger(rundir, 'cylc-reinstall')
+    REINSTALL_LOG.info(f"Reinstalling \"{named_run}\", from "
+                       f"\"{source}\" to \"{rundir}\"")
+    rsync_cmd = get_rsync_rund_cmd(
+        source, rundir, reinstall=True, dry_run=dry_run)
+    proc = Popen(rsync_cmd, stdout=PIPE, stderr=PIPE, universal_newlines=True)
```

Comment:
Had to look up `universal_newlines`, didn't know there was a shortcut to avoid calling `decode` all the time. Looks like it is deprecated by `text=True` since 3.7.

> The universal_newlines argument is equivalent to text and is provided for backwards compatibility. By default, file objects are opened in binary mode.